lib/resourcemerge/core.go

```diff
@@ -91,6 +61,38 @@ func ensurePodSpec(modified *bool, existing *corev1.PodSpec, required corev1.Pod
 	setInt32Ptr(modified, &existing.Priority, required.Priority)
 }
 
+func ensureContainers(modified *bool, existing *[]corev1.Container, required []corev1.Container) {
+	for i, existingContainer := range *existing {
+		matched := false
+		for _, requiredContainer := range required {
+			if existingContainer.Name == requiredContainer.Name {
+				matched = true
+				ensureContainer(modified, &existingContainer, requiredContainer)
+				break
+			}
+		}
+		if !matched {
+			*modified = true
+			*existing = append((*existing)[:i], (*existing)[i+1:]...)
+		}
+	}
+
+	for _, requiredContainer := range required {
+		var existingCurr *corev1.Container
+		for j, existingContainer := range *existing {
+			if existingContainer.Name == requiredContainer.Name {
+				existingCurr = &(*existing)[j]
+				ensureContainer(modified, existingCurr, requiredContainer)
```

Comment:
Hmm, Won't these containers already be ensured from previous loop??

Reply:
you are right. But above it dint work because of golang behavior where `requireContainer` is a local variable and changes to that does not update the slice `required`. Will fix, thanks!